### ISSUE TITLE
[Bug]: Shopify Theme Push CLI command not pushing / uploading changes to store using azure CI/CD

### ISSUE THREAD

**Issue by pp-gardinerbros** [Shopify/cli#3639]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Other

### Expected behavior

The shopify theme push command should show visual indication of changed files being uploaded in the command output. 

![image](https://github.com/Shopify/cli/assets/165791921/cab05414-0435-4a1a-b9f3-215a182df241)

### Actual behavior

We have azure CI/CD pipelines that have been pushing changes to multiple stores successfully last week. Over the last few days we have noticed the pipeline has not been uploading changes even though it is reporting the shopify theme push command completed successfully.

### Verbose output

```shell
2024-04-02T11:20:34.8307199Z 2024-04-02T11:20:34.814Z: Analytics event sent: {
2024-04-02T11:20:34.8307584Z   "command": "theme push",
2024-04-02T11:20:34.8307985Z   "time_start": 1712056832686,
2024-04-02T11:20:34.8308338Z   "time_end": 1712056834622,
2024-04-02T11:20:34.8308685Z   "total_time": 1936,
2024-04-02T11:20:34.8309007Z   "success": true,
2024-04-02T11:20:34.8310011Z   "cli_version": "3.58.0",
2024-04-02T11:20:34.8310351Z   "ruby_version": "3.1.4",
2024-04-02T11:20:34.8310693Z   "node_version": "18.19.1",
2024-04-02T11:20:34.8311028Z   "is_employee": false,
2024-04-02T11:20:34.8311360Z   "uname": "windows amd64",
2024-04-02T11:20:34.8311687Z   "env_ci": false,
2024-04-02T11:20:34.8312020Z   "env_plugin_installed_any_custom": false,
2024-04-02T11:20:34.8312473Z   "env_plugin_installed_shopify": "[\"@shopify/cli\",\"@shopify/plugin-did-you-mean\",\"@shopify/theme\"]",
2024-04-02T11:20:34.8312886Z   "env_shell": "cmd.exe",
2024-04-02T11:20:34.8313271Z   "env_device_id": "6d8ac9142c559c96a11ae8cde653f711a537bd71",
2024-04-02T11:20:34.8313632Z   "env_cloud": "localhost",
2024-04-02T11:20:34.8313977Z   "env_package_manager": "unknown",
2024-04-02T11:20:34.8314353Z   "cmd_all_timing_network_ms": 0,
2024-04-02T11:20:34.8314691Z   "cmd_all_timing_prompts_ms": 0,
2024-04-02T11:20:34.8315038Z   "cmd_all_launcher": "unknown",
2024-04-02T11:20:34.8315382Z   "cmd_all_topic": "theme",
2024-04-02T11:20:34.8315713Z   "cmd_all_plugin": "@shopify/theme",
2024-04-02T11:20:34.8316054Z   "cmd_all_force": false,
2024-04-02T11:20:34.8316585Z   "cmd_all_timing_active_ms": 1936,
2024-04-02T11:20:34.8316979Z   "cmd_all_exit": "ok",
2024-04-02T11:20:34.8317402Z   "args": "--store=saurabhtest-store.myshopify.com --theme 134863880376 --allow-live --json --verbose",
2024-04-02T11:20:34.8317899Z   "env_plugin_installed_all": "[\"@shopify/cli\",\"@shopify/plugin-did-you-mean\",\"@shopify/theme\"]",
2024-04-02T11:20:34.8318319Z   "metadata": "{\"extraPublic\":{},\"extraSensitive\":{}}"
2024-04-02T11:20:34.8318710Z }
2024-04-02T11:20:34.8319130Z 2024-04-02T11:20:34.814Z: Completed command theme push
```

### Reproduction steps

1. Run the command shopify theme push --store=xxxxxx.myshopify.com --theme xxxxxxxxxx --allow-live

### Operating System

Windows

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.58.0

### Shell

command

### Node version (run `node -v` if you're not sure)

18.19.1

### What language and version are you using in your application?

3.1.4

**Comment by mcarter-astronautdev** [comment]
We're having a similar new issue with `shopify theme push` within CI/CD using a Github Action .yml script.  We have several  Shopify themes packaged with Vitejs, all which have been successfully updated via the same Github action script for the last 9 months.  After building the theme with `npm run build`, our Github deployment action runs `shopify theme push --theme {{ NAME OF THEME }}` to push updates to our development preview theme on each site.  After the new CLI release last week, we discovered that our action script was suddenly pushing theme files that were missing our `main.css`, even though we can see in the action run log that the `main.css` file exists after the build is run.  Using `shopify theme push` manually in the terminal does not cause the same issue. 

Github Action:
`name: Develop | Build -> Deploy

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "main" branch
  push:
    branches: [ "develop" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  build:
    name: Build For Development
    if: ${{ !contains(github.event.head_commit.message, 'Update from Shopify') }}
    # The type of runner that the job will run on
    runs-on: ubuntu-latest
    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v4.1.1
      - uses: actions/setup-node@v4.0.2
        with:
          node-version: 18
      - uses: ruby/setup-ruby@v1.173.0
        with: 
          ruby-version: 3.1
          bundler: 'latest'
      # Installs the latest version of the shopify CLI
      - name: Install Shopify CLI
        run: npm install -g @shopify/cli @shopify/theme
      # Installs all dependancies in the package.json file
      - name: Install Dependancies
        run: npm i
      # Runs the Build script from package.json to build the theme files
      - name: Build and Package Theme Files.
        run: npm run build
      # Deploys code to shopify staging theme.
      - name: Deploy Development Theme.
        env:
          # Store URL, like your-store.myshopify.com
          SHOPIFY_FLAG_STORE: '${{ secrets.SHOPIFY_STORE }}'
          # Password generated from Theme Access app
          SHOPIFY_CLI_THEME_TOKEN: ${{ secrets.SHOPIFY_CLI_THEME_TOKEN }}
          SHOPIFY_CLI_TTY: 0
        run: shopify theme push --theme {{ theme-name-here }}
`

**Comment by nboliver-ventureweb** [comment]
@mcarter-astronautdev Looks like it's fixed in latest version of the CLI now, see https://github.com/Shopify/cli/issues/3616
